concourse/pipelines/checker-bot.yml

```diff
@@ -0,0 +1,71 @@
+---
+roles:
+  - name: owner
+    github:
+      teams: ["haiku:infrastructure"]
+    local:
+      users: ["admin"]
+resource_types:
+- name: gerrit
+  type: registry-image
+  source:
+    repository: docker.io/bluedocks/gerrit-resource
+resources:
+- name: haiku-gerrit
+  type: gerrit
+  source:
+    url: "https://review.haiku-os.org"
+    query: "status:open project:haiku"
+
+# Jobs
+jobs:
+- name: checker-bot
+  public: true
+  plan:
+    - get: haiku-gerrit 
+      trigger: true 
+    - task: check-coding-style
+      config:
+        inputs:
+        - name: haiku-gerrit
+        platform: linux
+        image_resource:
+          type: registry-image
+          source:
+            repository: docker.io/bluedocks/general-worker
+        run:
+          path: /bin/sh
+          args:
+            - -c
+            - |
+              cd ./haiku-gerrit
+              git config color.diff false
+              git clang-format --diff --style=haiku -q HEAD~1 > diff.txt
+              cat diff.txt
+
+              # check if diff.txt is empty or not and set verified
+              if [ -s diff.txt ]
+              then
+                verified=-1
+              else
+                verified=1
+              fi
+              echo "verified: $verified"
+
+              BASE_URL=https://review.haiku-os.org
+              CHANGE_ID=$(jq -r '.change_id' ./.gerrit_version.json)
+              REVISION_ID=$(jq -r '.revision' ./.gerrit_version.json)
+              TOKEN=Y2hlY2tlcmJvdDpnUlBONXdQQ0tVOTIxTUNzNitBc1dXclNSYzd4Mm9wWlNXNzNtYTRUOFE=
```

Comment:
We need to create a gerrit account for Checker Bot and replace its basic auth token here

Reply:
aahhhh.. this picks up on the Change ID.  Nice!   It does only seem to check style once the commit is in master though?

Reply:
https://github.com/haiku/infrastructure/blob/5f1bcbe80bc29ce20cdb9d76bae35df12db5ce49/concourse/pipelines/checker-bot.yml#L13-L18

Here `status:open project:haiku` specify the changeset/patchset which are going to be tracked. Any new changeset/patchset in the haiku project with open status will trigger this pipeline. Although it will not trigger when changes are merged to master branch.

I see in our gerrit code review website, the default search query is `status:open -is:wip`
What do you suggest to put in the query?